src/v0/destinations/impact_radius/transform.js

```diff
@@ -0,0 +1,285 @@
+      payload.AndroidId = get(message, "context.device.id");
+      payload.GoogAId = get(message, "context.device.advertisingId");
+    }
+    const productProperties = populateProductProperties(
+      productsMapping,
+      message.properties
+    );
+    const additionalParameters = populateAdditionalParameters(
+      message,
+      rudderToImpactProperty
+    );
+    if (eventType === "install" && payload.ClickId) {
+      delete payload.ClickId;
+    }
+    payload = { ...payload, ...productProperties, ...additionalParameters };
+
+    const endpoint = `${CONFIG_CATEGORIES.CONVERSION.base_url}${Config.accountSID}/Conversions`;
+    return responseBuilder(payload, endpoint, Config);
+  }
+  return responseBuilder(
+    buildPageLoadResponse(message, campaignId, impactAppId),
+    CONFIG_CATEGORIES.PAGELOAD.endPoint,
+    Config
+  );
+};
+
+/**
+ * This function is used to build the response for page call.
+ * @param {*} message
+ * @param {*} Config
+ * @returns
+ */
+const pageResponseBuilder = (message, Config) => {
+  const {
+    campaignId,
+    enablePageEvents,
+    impactAppId,
+    enableEmailHashing
+  } = Config;
+
+  if (!enablePageEvents) {
+    throw new ConfigurationError(
+      `${message.type} events are disabled from Config`
+    );
+  }
+
+  return responseBuilder(
+    buildPageLoadResponse(message, campaignId, impactAppId, enableEmailHashing),
+    CONFIG_CATEGORIES.PAGELOAD.endPoint,
+    Config
+  );
+};
+
+/**
+ * This function is used to build the response for screen call.
+ * @param {*} message
+ * @param {*} Config
+ * @returns
+ */
+const screenResponseBuilder = (message, Config) => {
+  const {
+    campaignId,
+    enableScreenEvents,
+    impactAppId,
+    enableEmailHashing
+  } = Config;
+
+  if (!enableScreenEvents) {
+    throw new ConfigurationError(
+      `${message.type} events are disabled from Config`
+    );
+  }
+
+  return responseBuilder(
+    buildPageLoadResponse(message, campaignId, impactAppId, enableEmailHashing),
+    CONFIG_CATEGORIES.PAGELOAD.endPoint,
+    Config
+  );
+};
+
+const processEvent = async (message, destination) => {
+  if (!message.type) {
+    throw new InstrumentationError("Event type is required");
+  }
+  if (checkConfigurationError(destination.Config)) {
+    const configField = checkConfigurationError(destination.Config);
+    throw new ConfigurationError(`${configField} is a required field`);
+  }
+
+  const messageType = message.type?.toLowerCase();
+  let response;
+  switch (messageType) {
+    case EventType.IDENTIFY:
+      response = identifyResponseBuilder(message, destination.Config);
+      break;
+    case EventType.TRACK:
+      response = trackResponseBuilder(message, destination.Config);
+      break;
+    case EventType.PAGE:
+      response = pageResponseBuilder(message, destination.Config);
+      break;
+    case EventType.SCREEN:
+      response = screenResponseBuilder(message, destination.Config);
```

Comment:
You can merge `screenResponseBuilder` and `trackResponseBuilder` and make it `trackAndScreenResponseBuilder`

Reply:
Actually, we have a different use cases for `track` and `screen` here

Reply:
My bad, It's not `trackResponseBuilder`, but `pageResponseBuilder`.

Reply:
Actually, we are doing the same thing for `identify`, `page` and `screen`. Just to keep the code readable I did it that way. We can club it together.